server/src/main/java/org/elasticsearch/action/search/SearchShard.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.action.search;
+
+import org.elasticsearch.common.Nullable;
+import org.elasticsearch.index.shard.ShardId;
+
+import java.util.Comparator;
+import java.util.Objects;
+
+/**
+ * A class that encapsulates the {@link ShardId} and the cluster alias
+ * of a shard used during the search action.
+ */
+public class SearchShard implements Comparable<SearchShard> {
```

Comment:
I wish that we could find a better name for this class. I find SearchShard confusing for some reason. But I am struggling to come up with something that's less confusing. I will propose something if I get ideas.